### PR TITLE
add project

### PR DESCRIPTION
**Pull request by cadohopi**

### CODE REVIEW

File: src/imovie.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// IMovie.ts
+import Genre from "./genre";
+
+interface IMovie {
```
Comment (kahnext) [inline src/imovie.ts:4]: theoretically, interface should be used to defined set up unique method, behavior but not include any method just to get property/attribute or even attribute definition

File: src/media.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Media.ts
+class Media {
+  constructor(
+    public title: string,
```
Comment (kahnext) [inline src/media.ts]: why do we define public keyword within a constructor?

Comment (kahnext) [inline src/media.ts]: where is Media's attribute?

File: src/movie.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  setRating(rating: number): void {
+    this.rating = rating;
+  }
+
+  getRating(): number {
+    return this.rating;
+  }
+
+  getDetails(): string {
+    return `- ${super.getDetails()} - Genre: ${this.genre.name} - Rating: ${this.getRating() || "Not rated yet"}`;
```
Comment (kahnext) [inline src/movie.ts]: why do we have dash character at the beginning?

File: src/movie_collection.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+// MovieCollection.ts
+import IMovie from './imovie';
+import IMovieCollection from './imovie_collection';
+
+class MovieCollection implements IMovieCollection {
```
Comment (kahnext) [inline src/movie_collection.ts:5]: where is the constructor? And what is this class used for?

File: src/genre.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+// Genre.ts
+class Genre {
+  constructor(public name: string, public description: string) {}
```
Comment (kahnext) [inline src/genre.ts]: why do we define scope keyword inside a constructor?

File: src/movie.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  setRating(rating: number): void {
+    this.rating = rating;
+  }
+
+  getRating(): number {
+    return this.rating;
+  }
+
```
Comment (kahnext) [inline src/movie.ts:22]: don't see any overloading implementation

File: src/movie_collection.ts (hunk 4 above)
Comment (cadohopi) [reply src/movie_collection.ts:5]: This class used to manage a collection of movies. It acts as a container or manager for multiple Movie objects.

File: src/movie.ts (hunk 3 above)
Comment (cadohopi) [reply src/movie.ts]: It starts a line in the list

**Pull request closed without merging**